src/zproxy.c

```diff
@@ -127,6 +133,12 @@ s_self_create_socket (self_t *self, char *type_name, char *endpoints, proxy_sock
             // Enable curve authentication
             zsock_set_curve_server (sock, 1);
         }
+
+        if (heartbeat) {
```

Comment:
this check is always true regardless of the content, which means the parameters are always set regardless of whether they are passed or not, which is not right

Reply:
You are right, thank you, I changed to checking the `ivl` field, if that is zero no other option matters.